pkg/agent/docker_secrets_postrenderer.go

```diff
@@ -52,6 +52,57 @@ func NewDockerSecretsPostRenderer(secrets map[string]string) (*DockerSecretsPost
 	return r, nil
 }
 
+// resourceListTypes defines all the list types for which we need to
+// search for pod templates.
+var resourceListTypes map[string]struct{} = map[string]struct{}{
+	"CronJobList":               struct{}{},
+	"DaemonSetList":             struct{}{},
+	"DeploymentList":            struct{}{},
+	"JobList":                   struct{}{},
+	"PodTemplateList":           struct{}{},
+	"ReplicaSetList":            struct{}{},
+	"ReplicationControllerList": struct{}{},
+	"StatefulSetList":           struct{}{},
+}
+
+func (r *DockerSecretsPostRenderer) processResourceList(resourceList []interface{}) {
+	for _, resourceItem := range resourceList {
+		resource, ok := resourceItem.(map[interface{}]interface{})
+		if !ok {
+			continue
+		}
+		kindValue, ok := resource["kind"]
+		if !ok {
+			log.Errorf("invalid resource: no kind. %+v", resource)
+			continue
+		}
+
+		kind, ok := kindValue.(string)
+		if !ok {
+			log.Errorf("invalid resource: non-string resource kind. %+v", resource)
+			continue
+		}
+		if _, ok := resourceListTypes[kind]; ok {
+			if items, ok := resource["items"]; ok {
```

Comment:
can't we just check if the resource has the key `"items"` to know if it's a list? That may be a slightly slower (since we would be processing list for other than workloads) but safer and we would need to maintain less hard-coded types.

Reply:
We could, and it would work now. I thought it better to be explicit, since we don't control the API, but perhaps it would be simpler. Updating.